### PR TITLE
Clean up forking and messages

### PR DESCRIPTION
**Pull request by vdemedes**
This is a clean up kind of PR, not refactoring.

### CODE REVIEW

File: lib/runner.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 'use strict';
-var util = require('util');
 var EventEmitter = require('events').EventEmitter;
 var Promise = require('bluebird');
 var hasFlag = require('has-flag');
+var util = require('util');
```
Comment (sindresorhus) [inline lib/runner.js]: I prefer to have builtin requires grouped first.

File: lib/babel.js [hunk 2]
```diff
@@ -16,64 +20,69 @@ try {
...
+	if (!message.ava) {
+		return;
 	}
+
+	process.emit(message.name, message.data);
 });
 
-process.on('ava-kill', function () {
+process.on('ava-exit', function () {
+	// use a little delay when running on AppVeyor (reason unknown)
```
Comment (sindresorhus) [inline lib/babel.js]: We know the reason, AppVeyor is just super bad and flaky :p

File: lib/fork.js [hunk 3]
```diff
@@ -1,54 +1,43 @@
 'use strict';
 var childProcess = require('child_process');
 var Promise = require('bluebird');
-var path = require('path');
+var dirname = require('path').dirname;
+var join = require('path').join;
```
Comment (sindresorhus) [inline lib/fork.js]: Can you just import `path` instead and use methods on it?

Comment (vdemedes) [inline lib/fork.js]: yeah, did that before, but we need `path` here - https://github.com/sindresorhus/ava/pull/236/files#diff-a48173105195b7a6481a3ad85b45144eR13

When we will support #229, `babel` variable name won't be a good choice.

File: lib/babel.js (hunk 2 above)
Comment (vdemedes) [inline lib/babel.js]: will update the comment ;)

File: lib/runner.js (hunk 1 above)
Comment (vdemedes) [inline lib/runner.js]: Np ;) I just order stuff by module name length in my projects.

File: lib/fork.js [hunk 4]
```diff
@@ -56,20 +45,36 @@ module.exports = function (args) {
...
 		});
 	});
 
 	// emit `test` and `stats` events
 	ps.on('message', function (event) {
+		if (!event.ava) {
+			return;
+		}
+
+		event.name = event.name.replace('ava-', '');
```
Comment (sindresorhus) [inline lib/fork.js]: `'ava-'` → `/^ava-/` just to be sure we don't strip any future message with `ava-` in the middle. I know, edge case, but this gives clearer intent too.

Comment (vdemedes) [inline lib/fork.js]: Ok!

File: lib/send.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+// utility to send messages to processes
+function send(ps, name, data) {
+	if (typeof ps === 'string') {
```
Comment (sindresorhus) [inline lib/send.js]: Is this really needed? Do we really pass a custom `process` anywhere? And even if we do, wouldn't it be better to just have it as the third argument instead? For simplicity.

Comment (vdemedes) [inline lib/send.js]: We pass messages in `lib/fork.js`. Yeah, I can move it to 3rd argument.

File: lib/fork.js (hunk 3 above)
Comment (sindresorhus) [inline lib/fork.js]: Call it `filepath` or `pth`.

File: test/fixture/long-running.js [hunk 6]
```diff
@@ -11,7 +11,11 @@ test('long running', function (t) {
 		while(Date.now() - start < 2000) {
 			//synchronously wait for 2 seconds
 		}
-		process.send({name:'cleanup-completed', data: {completed: true}});
+		process.send({
+			name:'cleanup-completed',
```
Comment (sindresorhus) [inline test/fixture/long-running.js]: needs a space after `:`

**Comment by sindresorhus** [comment]
// @jamestalmage

File: lib/send.js (hunk 5 above)
Comment (vdemedes) [inline lib/send.js]: Turns out it will conflict with the following:

```js
send('teardown', ps);
```

where events don't have any data.

Comment (sindresorhus) [inline lib/send.js]: Oh, ok, nvm then.

**Comment by vdemedes** [comment]
@jamestalmage thanks for the tip!

**Comment by jamestalmage** [comment]
Nice!

You taking a break now, or are you moving on to a major refactor?

**Comment by vdemedes** [comment]
No no, I want to clean up everything, all files and tests, then move on to refactoring.

**Comment by jamestalmage** [comment]
Leave `runner.js` to me for a while if you don't mind.

**Comment by vdemedes** [comment]
Unfortunately, code interferes between files, so I can't guarantee I can leave unaffected runner.js. I was hoping to do it all in one wave, so that I don't wait for anyone else (that is why we cleaned up our PR queue) and after that, any suggestions and contributions are welcome ;)

**Comment by jamestalmage** [comment]
OK

**Pull request merged**